cmd/hadolintExecute.go

```diff
@@ -0,0 +1,130 @@
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+)
+
+// HadolintPiperFileUtils ..
+// mock generated with: mockery --name HadolintPiperFileUtils --dir cmd --output pkg/hadolint/mocks
+type HadolintPiperFileUtils interface {
+	FileExists(filename string) (bool, error)
+	FileWrite(filename string, data []byte, perm os.FileMode) error
+}
+
+// HadolintClient ..
+// mock generated with: mockery --name hadolintClient --dir cmd --output pkg/hadolint/mocks
+type HadolintClient interface {
+	SetOptions(options piperhttp.ClientOptions)
+	DownloadFile(url, filename string, header http.Header, cookies []*http.Cookie) error
+}
+
+type hadolintRunner interface {
+	RunExecutable(executable string, params ...string) error
+	Stdout(err io.Writer)
+	Stderr(err io.Writer)
+}
+
+type hadolintUtils struct {
+	HadolintPiperFileUtils
+	HadolintClient
+	hadolintRunner
+}
+
+func hadolintExecute(config hadolintExecuteOptions, _ *telemetry.CustomData) {
+	runner := command.Command{
+		ErrorCategoryMapping: map[string][]string{},
+	}
+	// reroute runner output to logging framework
+	runner.Stdout(log.Writer())
+	runner.Stderr(log.Writer())
+
+	utils := hadolintUtils{
+		HadolintPiperFileUtils: &piperutils.Files{},
+		HadolintClient:         &piperhttp.Client{},
+		hadolintRunner:         &runner,
+	}
+
+	if err := runHadolint(config, utils); err != nil {
+		log.Entry().WithError(err).Fatal("Execution failed")
+	}
+}
+
+func runHadolint(config hadolintExecuteOptions, utils hadolintUtils) error {
+	var outputBuffer bytes.Buffer
+	var errorBuffer bytes.Buffer
+	utils.Stdout(&outputBuffer)
+	utils.Stderr(&errorBuffer)
+
+	clientOptions := piperhttp.ClientOptions{
+		TransportTimeout:          20 * time.Second,
+		TransportSkipVerification: true,
+	}
+	if len(config.ConfigurationUsername) > 0 {
+		clientOptions.Username = config.ConfigurationUsername
+		clientOptions.Password = config.ConfigurationPassword
+	}
+	utils.SetOptions(clientOptions)
+
+	options := []string{
+		"--format checkstyle",
+	}
+	// load config file from URL
+	if !hasConfigurationFile(config.ConfigurationFile, utils) && len(config.ConfigurationURL) > 0 {
+		if err := loadConfigurationFile(config.ConfigurationURL, config.ConfigurationFile, utils); err != nil {
+			return errors.Wrap(err, "failed to load configuration file from URL")
+		}
+	}
+	// use config
+	if hasConfigurationFile(config.ConfigurationFile, utils) {
+		options = append(options, fmt.Sprintf("--config %s", config.ConfigurationFile))
+		log.Entry().WithField("file", config.ConfigurationFile).Debug("Using configuration file")
+	} else {
+		log.Entry().Debug("No configuration file found.")
+	}
+	// execute scan command
+	runCommand := fmt.Sprintf("hadolint %s %s", config.DockerFile, strings.Join(options, " "))
+	runCommandTokens := tokenize(runCommand)
```

Comment:
any reason why you first create a string and then tokenize it again?
Couldn't you just add `config.Dockerfile` as first option and then pass it to `RunExecutable`?

Reply:
No, seems to be a leftover, I will remove it.

Reply:
One doubt, it aliasing for step parameter supported or only for the config file?

Reply:
Looks like I need the `tokenize` to split the option and it's value `--format checkstyle`.

Reply:
removed